### PR TITLE
Enable Thread Runner in AT

### PR DESCRIPTION
**Pull request by rain-on**
Also includes some basic refactorings to the AT in general.

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  private final SignerConfiguration signerConfig;
+  private final Path dataPath;
+  private final Properties portsProperties;
+
+  private static final String PORTS_FILENAME = "eth2signer.ports";
+  private static final String HTTP_PORT_KEY = "http-port";
+
+  public static Eth2SignerRunner createRunner(final SignerConfiguration signerConfig) {
+    if (Boolean.getBoolean("acctests.runEth2SignerAsProcess")) {
+      return new Eth2SignerProcessRunner(signerConfig);
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java]: Think would be useful to log which runner we are using

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  public Eth2SignerProcessRunner(final SignerConfiguration signerConfig) {
+    super(signerConfig);
+  }
+
+  @Override
+  protected void startExecutor(final List<String> params) {
+    final StringJoiner javaOpts = new StringJoiner(" ");
+
+    final String[] paramsAsArray = params.toArray(new String[params.size()]);
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java]: Intellij is recommending to use toArray without sizing i.e. new String[0]. Their rationale seems sensible enough so think we go with that.

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            .redirectInput(Redirect.INHERIT);
+
+    if (Boolean.getBoolean("debugSubProcess")) {
+      javaOpts.add("-Xdebug -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005");
+    }
+    processBuilder.environment().put("JAVA_OPTS", javaOpts.toString());
+
+    try {
+      process = processBuilder.start();
+      outputProcessorExecutor.submit(() -> printOutput());
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java]: use method reference for printOutput

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/Signer.java [hunk 4]
```diff
@@ -30,31 +32,32 @@
...
   private final String hostname;
   private final String urlFormatting;
   private final Vertx vertx;
   private HttpClient httpClient;
 
   public Signer(final SignerConfiguration signerConfig) {
-    this.runner = new Eth2SignerProcessRunner(signerConfig);
+    this.runner = Eth2SignerRunner.createRunner(signerConfig);
     this.hostname = signerConfig.hostname();
     urlFormatting = "http://%s:%s";
```
Comment (usmansaleem) [inline acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/Signer.java]: nit: the url formatting can be a constant.

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    params.add("--logging");
+    params.add(loggingLevel);
+    params.add("--http-listen-host");
+    params.add(signerConfig.hostname());
+    params.add("--http-listen-port");
+    params.add(String.valueOf(signerConfig.httpPort()));
+    params.add("--key-store-path");
+    params.add(signerConfig.getKeyStorePath().toString());
+    if (signerConfig.isDynamicPortAllocation()) {
+      params.add("--data-path");
```
Comment (usmansaleem) [inline acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java]: question, we don't need datapath if we are not using dynamic ports?

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    final File portsFile = new File(dataPath.toFile(), PORTS_FILENAME);
+    LOG.info("Awaiting presence of ethsigner.ports file: {}", portsFile.getAbsolutePath());
+    awaitPortsFile(dataPath);
+    LOG.info("Found ethsigner.ports file: {}", portsFile.getAbsolutePath());
+
+    try (final FileInputStream fis = new FileInputStream(portsFile)) {
+      portsProperties.load(fis);
+      LOG.info("EthSigner ports: {}", portsProperties);
+    } catch (final IOException e) {
+      throw new RuntimeException("Error reading Web3Provider ports file", e);
```
Comment (usmansaleem) [inline acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java]: nit: There is `UncheckedIOException` in standard JDK which is designed to convert IOexception to runtime exception.

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/Signer.java (hunk 4 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/Signer.java]: done

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java (hunk 1 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java]: done

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java (hunk 2 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java]: done

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java (hunk 3 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerProcessRunner.java]: done

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java (hunk 5 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java]: correct - datapath is where we put our ports.ethsigner file - but if we're using a fixed port, no need to track it - this is historical.

**Pull request merged**